src/authorizations/pagination/TokensTab.tsx

```diff
@@ -122,46 +132,79 @@ class TokensTab extends PureComponent<Props, State> {
       rightHeaderItems = <GenerateTokenDropdownRedesigned />
     }
 
+    const tokensBanner = () => {
+      if (!isFlagEnabled('tokensUIRedesign')) {
+        return (
+          <>
+            <BannerPanel
+              size={ComponentSize.ExtraSmall}
+              gradient={Gradients.PolarExpress}
+              icon={IconFont.Bell}
+              hideMobileIcon={true}
+              textColor={InfluxColors.Yeti}
+            >
+              <TokensRedesignBanner />
+            </BannerPanel>
+          </>
+        )
+      }
+    }
+
     return (
-      <AutoSizer>
-        {({width, height}) => {
-          const heightWithPagination =
-            this.paginationRef?.current?.clientHeight +
-              DEFAULT_TAB_NAVIGATION_HEIGHT ||
-            DEFAULT_PAGINATION_CONTROL_HEIGHT + DEFAULT_TAB_NAVIGATION_HEIGHT
-
-          const adjustedHeight = height - heightWithPagination
-          return (
-            <>
-              <TabbedPageHeader
-                childrenLeft={leftHeaderItems}
-                childrenRight={rightHeaderItems}
-                width={width}
-              />
-              <FilterAuthorizations
-                list={tokens}
-                searchTerm={searchTerm}
-                searchKeys={this.searchKeys}
-              >
-                {filteredAuths => (
-                  <TokenList
-                    tokenCount={tokens.length}
-                    auths={filteredAuths}
-                    emptyState={this.emptyState}
-                    pageWidth={width}
-                    pageHeight={adjustedHeight}
-                    searchTerm={searchTerm}
-                    sortKey={sortKey}
-                    sortDirection={sortDirection}
-                    sortType={sortType}
-                    onClickColumn={this.handleClickColumn}
+      <>
+        {tokensBanner()}
+        <AutoSizer>
+          {({width, height}) => {
+            // if tokens redesign flag is off, adjust the page height so the banner doesn't push the pagination controller off
+            let heightWithPagination
+            isFlagEnabled('tokensUIRedesign')
+              ? (heightWithPagination =
+                  this.paginationRef?.current?.clientHeight +
+                    DEFAULT_TAB_NAVIGATION_HEIGHT ||
+                  DEFAULT_PAGINATION_CONTROL_HEIGHT +
+                    DEFAULT_TAB_NAVIGATION_HEIGHT)
+              : (heightWithPagination =
+                  this.paginationRef?.current?.clientHeight +
+                    DEFAULT_TAB_NAVIGATION_HEIGHT ||
+                  DEFAULT_PAGINATION_CONTROL_HEIGHT +
+                    DEFAULT_TAB_NAVIGATION_HEIGHT +
+                    DEFAULT_ALERT_HEIGHT)
```

Comment:
this is totally passable as is, but in the future, don't be afraid to pull these calculations out of the `render` method into an `if/else` block and store the values in variables if they start getting complex. it's a blancing act between having to refer to a variable defined elsewhere, and having to reason about expressions that have a lot of branching logic

Reply:
yeah this one got a little too wordy and unnecessarily long. But since we will be restoring the code back to how it was (after Jan 2nd), i figured it will do for now.